### ISSUE TITLE
[BUG] Rapidgator utterly broken :(

### ISSUE THREAD

**Issue by FunsKiTo** [Official-Husko/mul-tor#25]
~The upload fails if the filename contains "[]"~

- Uploading a single file to Rapidgator works fine.
- However, if I choose multiple files option, it fails and gives me the following error (I tried with both 1 file and with multiple files in the folder):
`[Errno 2] No such file or directory: 'file.rar'`

Additionally, I keep encountering this exception at random with Rapidgator:
`Exception: Login Failed? Please check your account details else report this.`

**Comment by Official-Husko** [comment]
That's good to know thank you. Will patch it in a hotfix quickly along the other issues.

**Comment by Official-Husko** [comment]
Wait have you been able to upload without the "[]" characters in the name? Currently support for this site is in Experimental due to some issues i could never seem to resolve.

**Comment by FunsKiTo** [comment]
> Wait have you been able to upload without the "[]" characters in the name? Currently support for this site is in Experimental due to some issues i could never seem to resolve.

Yes
![image](https://github.com/Official-Husko/mul-tor/assets/97408428/2027c2d7-7c4d-44d1-8a35-05768fac7718)

**Comment by Official-Husko** [comment]
> > Wait have you been able to upload without the "[]" characters in the name? Currently support for this site is in Experimental due to some issues i could never seem to resolve.
> 
> Yes ![image](https://private-user-images.githubusercontent.com/97408428/322116437-2027c2d7-7c4d-44d1-8a35-05768fac7718.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI5NTA5NTUsIm5iZiI6MTcxMjk1MDY1NSwicGF0aCI6Ii85NzQwODQyOC8zMjIxMTY0MzctMjAyN2MyZDctN2M0ZC00NGQxLThhMzUtMDU3NjhmYWM3NzE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDEyVDE5MzczNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxY2FlNjg3ZTcyNjVjYWFlNDUzOTI0YTBiYTliYTNlYzBiZWZhZTgzODZmMWFjZmMzNzhiNTVmNDVlOGM0NzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NE8SB_l6Ig27EGzvb0Uxo0nSSMTuJgwAA5VrvUUnTwo)

Interesting.. If you should face any issues please let me know because if not then it seems to be working as intended.

**Comment by Official-Husko** [comment]
> The upload fails if the filename contains "[]"

After further investigation it seems that "[]" is not an issue. Could you share the full file name with me?

**Comment by FunsKiTo** [comment]
> > The upload fails if the filename contains "[]"
> 
> After further investigation it seems that "[]" is not an issue. Could you share the full file name with me?

You're right, it's working now. Maybe it had nothing to do with the name, it just started working the moment I tried without "[]". Sorry.

**Comment by Official-Husko** [comment]
> > > The upload fails if the filename contains "[]"
> > 
> > 
> > After further investigation it seems that "[]" is not an issue. Could you share the full file name with me?
> 
> You're right, it's working now. Maybe it had nothing to do with the name, it just started working the moment I tried without "[]". Sorry.

It is very possible that you had one of these weird anomalies i have had with this site. If encounter it again try to give me the full file name and the error message or whatever it spits at you.

**Comment by FunsKiTo** [comment]
After further testing in v1.5.1, I encountered this issue. Oh, and it has nothing to do with the filename:

- Uploading a single file to Rapidgator works fine.
- However, if I choose multiple files option, it fails and gives me the following error (I tried with both 1 file and with multiple files in the folder):
[Errno 2] No such file or directory: 'file.rar'

Additionally, regarding Mixdrop after the update, I encountered the following error:
'NoneType' object has no attribute 'get'

Lastly, Buzzheavier works, but it exhibits the same problem as PixelDrain with low speeds. I'm unsure if I should open another report for this.

Thanks for the quick fixes and for the amazing tool.

**Comment by FunsKiTo** [comment]
Actually, I keep encountering this exception at random:
`Exception: Login Failed? Please check your account details else report this.`

**Comment by Official-Husko** [comment]
> After further testing in v1.5.1, I encountered this issue. Oh, and it has nothing to do with the filename:
> 
>     * Uploading a single file to Rapidgator works fine.
> 
>     * However, if I choose multiple files option, it fails and gives me the following error (I tried with both 1 file and with multiple files in the folder):
>       `[Errno 2] No such file or directory: 'file.rar'`
>       In 1.5.0 was also giving me:
>       `Exception: Login Failed? Please check your account details else report this.`
>       Doesn't seem to happen in 1.5.1
> 
> 
> Additionally, regarding Mixdrop after the update, I encountered the following error: 'NoneType' object has no attribute 'get'
> 
> Lastly, Buzzheavier works, but it exhibits the same problem as PixelDrain with low speeds. I'm unsure if I should open another report for this.
> 
> Thanks for the quick fixes and for the amazing tool.

The issue you are facing with the "no file" on Rapidgator is the exact issue i faced and even tho it should be dead simple to fix i was not able to solve it somehow.

I'm investigating Mixdrop right now to see if i can recreate the issue. They changed their system not long ago so its possible someone touched something again there which broke it all.

about the speed issue, i will probably do a full update on all supported sites tomorrow to make them all faster. Have you used Pixeldrain by any chance to see if the speed issue has been resolved?

> Actually, I keep encountering this exception at random with RapidGator: `Exception: Login Failed? Please check your account details else report this.`

Sounds about right.. Rapidgator always has been a massive bitch somehow with their shit. It is very likely that it has to do with some api login limit which im not not or at least wasn't aware of and which the program is currently not handling. I will do some additional testing to see if i can figure out why. For

**Comment by FunsKiTo** [comment]
> Have you used Pixeldrain by any chance to see if the speed issue has been resolved?
Yes, I'm on a 1GB connection, and it gives full upload speed (way faster than with the browser).

> Edit: I took the liberty to update the description for the new issues raised and the title to keep my sanity alive a bit longer. I have also opened a few extra issues with the problems you encountered so i can properly address them tomorrow.
Sorry, that one is on me. I'm new to reporting problems on GitHub. It makes sense to have the issues individually organized.

**Comment by Official-Husko** [comment]
> > Have you used Pixeldrain by any chance to see if the speed issue has been resolved?
> 
> Yes, I'm on a 1GB connection, and it gives full upload speed (way faster than with the browser).
> 

Perfect. I will update the rest then too.

> > Edit: I took the liberty to update the description for the new issues raised and the title to keep my sanity alive a bit longer. I have also opened a few extra issues with the problems you encountered so i can properly address them tomorrow.
> 
> Sorry, that one is on me. I'm new to reporting problems on GitHub. It makes sense to have the issues individually organized.

All good, thank you for reporting the issues and using the tool! Glad to see people use and enjoy it.

**Comment by Official-Husko** [comment]
@FunsKiTo Please visit #29 for the fix and reason regarding Mixdrop.

**Comment by FunsKiTo** [comment]
> Please visit https://github.com/Official-Husko/mul-tor/issues/29 for the fix and reason regarding Mixdrop.

Working with the name change.
